src/docs/release/breaking-changes/use-maxLengthEnforcement-instead-of-maxLengthEnforced.md

```diff
@@ -0,0 +1,211 @@
+  * iOS, Web, MacOS, Linux, Fuchsia: defaults to
+    `MaxLengthEnforcement.truncateAfterCompositionEnds`.
+
+## Migration guide
+
+_Use the default behavior for the current platform is recommended, since it
+won't break someone that is not surposed to input like the behavior that
+developers has defined._
+
+### Default values of `maxLengthEnforcement`
+
+* Android, Windows: `MaxLengthEnforcement.enforced`. The native behavior of
+  these platforms is enforced. The composing will be handled by the IME while
+  users are entering CJK characters.
+* iOS: `MaxLengthEnforcement.truncateAfterCompositionEnds`. iOS has no default
+  behavior and it requires users implement the behavior themselves. Allow the
+  composition to exceed to avoid breaking CJK input.
+* Web, MacOS, Linux, Fuchsia: `MaxLengthEnforcement.truncateAfterCompositionEnds`.
+  These platforms allow the composition to exceed by default.
+
+### To enforce the limit all the time
+
+To enforce the limit that always truncate the value when it reaches the limit
+(e.g. entering verfication code), use `MaxLengthEnforcement.enforced` in
+editable text fields.
+
+_Be aware that in common cases, enforced truncate might make users failed to
+enter proposal CJK characters within `maxLength` limit on a phonetic keyboard,
+especially on platforms that the behavior defaults to
+`MaxLengthEnforcement.truncateAfterCompositionEnds`._
+
+Code before migration:
+
+<!-- skip -->
+```dart
+TextField(maxLength: 6)
+```
+
+or 
+
+<!-- skip -->
+```dart
+TextField(
+  maxLength: 6,
+  maxLengthEnforced: true,
+)
+```
+
+Code after migration:
+
+<!-- skip -->
+```dart
+TextField(
+  maxLength: 6,
+  maxLengthEnforcement: MaxLengthEnforcement.enforced,
+)
+```
+
+### To not enforce the limitation
+
+To show a max length error in `TextField`, but _not_ truncate when the limit
+is exceeded, use `MaxLengthEnforcement.none` instead of
+`maxLengthEnforced: false`.
+
+Code before migration:
+
+<!-- skip -->
+```dart
+TextField(
+  maxLength: 6,
+  maxLengthEnforced: false,
+)
+```
+
+Code after migration:
+
+<!-- skip -->
+```dart
+TextField(
+  maxLength: 6,
+  maxLengthEnforcement: MaxLengthEnforcement.none,
+)
+```
+
+For `CupertinoTextField`, which isn't able to show an error message,
+just don't set the `maxLength` value.
+
+Code before migration:
+
+<!-- skip -->
+```dart
+CupertinoTextField(
+  maxLength: 6,
+  maxLengthEnforced: false,
+)
+```
+
+Code after migration:
+
+<!-- skip -->
+```dart
+CupertinoTextField(
+  // Leave it not set.
+)
+```
+
+### To enforce the limit, but not for composing text
+
+To avoid truncating text while the user is inputting text via composition,
+specify `MaxLengthEnforcement.truncateAfterCompositionEnds`. This behavior
+allows some Chinese, Japanese, and Korean (CJK) characters to temporarily
+ignore the limit until editing is complete.
+
+_Gboard on Android produces the composing text when the user is entering Latin
+character/words by default._
```

Comment:
"Gboard's English keyboard on Android (and many other Android input methods) creates a composing region
for the current English word being entered. When used in a `truncateAfterCompositionEnds` text field, the
user may not be stopped right away at the `maxLength` limit. Consider the `enforced` option if the text field 
does not expect characters that may require multistage text input on a phonetic keyboard (e.g., CJK characters)."

Reply:
I'm not sure CJK scripts are the only ones that require multiple key presses to enter one character. I'd like to avoid using "multistage text input" since it's not very clear but can't find a better way to put it.

Reply:
Haven't found other inputs that related to multistage text input besides CJKs. I think we can use "CJK characters" directly because we use this sentence through the whole guide?

Reply:
> Consider the `enforced` option if the text field does not expect CJK characters on a phonetic keyboard.